### ISSUE TITLE
Make `QueryError` -> `CassError` conversion more accurate

### ISSUE THREAD

**Issue by wprzytula** [scylladb/cpp-rs-driver#133]
## Problem

At the moment, the error conversion is much limited due to stringified errors in Rust driver.
The biggest offender in my opinion is:
```rust
    QueryError::InvalidMessage(_string) => CassError::CASS_ERROR_SERVER_INVALID_QUERY,
```
which is totally inaccurate, because `SERVER_INVALID_QUERY` strictly corresponds to `DbError::Invalid`. At the same time, `QueryError::InvalidMessage` is abused so much by converting all `ParseError`s and `FrameError`s to it.

## Solution

Once the errors refactor lands in the Rust driver, a complex review of existing error conversion should be done to make it more accurate. Or else users moving from cpp-driver to cpp-rust-driver will get much confused by error messages. After all, failure deserializing a row may be displayed at the moment as "server invalid query".

**Comment by wprzytula** [comment]
Another finding:
`NewSessionError::ProtocolError` is converted to a different error than `QueryError::ProtocolError`.
```rust
    NewSessionError::ProtocolError(_str) => {
        CassError::CASS_ERROR_LIB_UNABLE_TO_DETERMINE_PROTOCOL
    }
```
```rust
    DbError::ProtocolError => CassError::CASS_ERROR_SERVER_PROTOCOL_ERROR,
```
This is a mess.

**Comment by Lorak-mmk** [comment]
Yeah, when writing the initial version during the hachathon we didn't put much care in error conversions. Getting something that mostly works was more important.
After error refactor in Rust Driver is finished will be a good time to review the conversions to see which errors cpp-driver returns under which conditions and convert accordingly.
